### DISCUSSION TITLE
fpPromise.then is not a function

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [fingerprintjs/fingerprintjs#744] [Q&A]
did someone facing the `fpPromise.then is not a function` error?

https://stackoverflow.com/questions/71181994/typeerror-fppromise-then-is-not-a-function

**Comment by makma** [comment]
Hi @jiangxiaoqiang ,
If you want to use the NPM package with `require` rather than our [CDN approach](https://github.com/fingerprintjs/fingerprintjs#quick-start), you need to call the `load()` function first.
```js
...
// Initialize an agent at application startup.
const fpPromise = require('@fingerprintjs/fingerprintjs').load();
// Get the visitor identifier when you need it.
fpPromise
    .then((fp) => fp.get())
    .then(async (result) => {
        // This is the visitor identifier:
        const deviceId = console.log(result.visitorId);
    });
...
```

**Reply by Finesse** [reply]
Here is a code example from [the documentation](https://github.com/fingerprintjs/fingerprintjs/blob/master/docs/api.md#webpackrollupnpmyarn):

```js
const FingerprintJS = require('@fingerprintjs/fingerprintjs')

// Initialize the agent at application startup.
const fpPromise = FingerprintJS.load()

// Get the visitor identifier when you need it.
fpPromise
  .then(fp => fp.get())
  .then(result => console.log(result.visitorId))
```

It's equivalent to the @makma's example.
